src/components/map/components/legend/index.js

```diff
@@ -0,0 +1,168 @@
+import { css, html, LitElement, unsafeCSS } from 'lit';
+import { VlMapVectorLayer } from '../layer/vector-layer/index.js';
+import styles from './styles.scss';
+import { LEGEND_PLACEMENT } from './enums/index.js';
+import { VlMapLayerCircleStyle } from '../layer-style/layer-circle-style/index.js';
+
+export class VlMapLegend extends LitElement {
+  static get styles() {
+    return [
+      css`
+        ${unsafeCSS(styles)}
+      `,
+    ];
+  }
+
+  constructor() {
+    super();
+    this.placement = LEGEND_PLACEMENT.BOTTOM_RIGHT;
+  }
+
+  static get properties() {
+    return {
+      left: { type: String, reflect: true },
+      top: { type: String, reflect: true },
+      right: { type: String, reflect: true },
+      bottom: { type: String, reflect: true },
+      placement: {
+        type: String,
+        attribute: 'data-vl-placement',
+        reflect: true,
+      },
+    };
+  }
+
+  __getPosition() {
+    const position = {};
+
+    switch (this.placement) {
+      case LEGEND_PLACEMENT.TOP_LEFT:
+        console.log('TOP_LEFT');
+        position.top = '10px';
+        position.left = '10px';
+        position.right = undefined;
+        position.bottom = undefined;
+        console.log({ position });
+        break;
+      case LEGEND_PLACEMENT.TOP_RIGHT:
+        console.log('TOP_RIGHT');
+        position.top = '10px';
+        position.left = undefined;
+        position.right = '10px';
+        position.bottom = undefined;
+        console.log({ position });
+        break;
+      case LEGEND_PLACEMENT.BOTTOM_LEFT:
+        console.log('BOTTOM_LEFT');
+        position.top = undefined;
+        position.left = '8px';
+        position.right = undefined;
+        position.bottom = '40px';
+        break;
+      case LEGEND_PLACEMENT.BOTTOM_RIGHT:
+        console.log('BOTTOM_RIGHT');
+        position.top = undefined;
+        position.left = undefined;
+        position.right = '58px';
+        position.bottom = '10px';
+        break;
+      default:
+        console.log(`Invalid placement ${this.placement}!!!`);
+        break;
+    }
+
+    console.log({ position });
+
+    console.log('top');
+    console.log(this._top);
+    if (this._top !== undefined) position.top = this._top;
+    if (this._left) position.left = this._left;
+    if (this._right) position.right = this._right;
+    if (this._bottom) position.bottom = this._bottom;
+
+    console.log({ position });
+
+    return position;
+  }
+
+  set left(left) {
+    this._left = left;
+  }
+
+  set right(right) {
+    this._right = right;
+  }
+
+  set top(top) {
+    this._top = top;
+  }
+
+  set bottom(bottom) {
+    this._bottom = bottom;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._mapElement = this.closest('vl-map');
+    this._featuresLayers = this._mapElement.featuresLayers;
```

Comment:
enkel voor features-layers is voldoende nu? voor wfs-lagen kan je vermoedelijk ook wel iets uit de metadata halen, voor image based (wms, wmts) kaarten wschl moeilijker ...

Reply:
aanpassing gedaan zodat de stijlen van een wfs laag nu ok in de legende getoond worden met de naam van de was laag